frontend/src/components/SystemsTable/SystemTableContent.tsx

```diff
@@ -220,6 +247,50 @@ export function SystemTableContent({
     },
   };
 
+  const handleTableChange = (
+    _pagination: TablePaginationConfig,
+    filters: Record<string, FilterValue | null>,
+    sorter: SorterResult<SystemModel> | SorterResult<SystemModel>[]
+  ) => {
+    // Handle filter change
+    for (const k in filters) {
+      if (
+        k === "dataset.split" &&
+        filters[k]?.toString() !== filterValue.split
+      ) {
+        onFilterChange({ split: filters[k]?.toString() });
+      } else if (k === "task" && filters[k]?.toString() !== filterValue.task) {
```

Comment:
Looks like `filters[k]` can sometimes be an array so this does not work properly. I would suggest to avoid `toString()` as a type cast.

Reply:
Now the filters only allow one filtered value, so `filters[k]` is of type `FilterValue | null`. `FilterValue` is defined as `React.Key | boolean`, but the `split` is of type `string | undefined`. 

In PR #528 , I try to separate the `null/undefined` and `Key/string` cases, but I don't find a way to avoid type casting.